assets/snippets/docinfo/snippet.docinfo.php

```diff
@@ -18,10 +19,22 @@
 *       [[DocInfo? &docid=`15` &field=`pagetitle`]]
 *       [[DocInfo? &docid=`10` &field=`tvname`]]
 *       [[DocInfo? &docid=`3` &field=`tvname` &render=`1`]]
+*       [[DocInfo? &docid=`3` &templid=`2` &field=`tvname`]]
 */
 if(!defined('MODX_BASE_PATH')){die('What are you doing? Get out of here!');}
 $default_field = array('type','contentType','pagetitle','longtitle','description','alias','link_attributes','published','pub_date','unpub_date','parent','isfolder','introtext','content','richtext','template','menuindex','searchable','cacheable','createdon','createdby','editedon','editedby','deleted','deletedon','deletedby','publishedon','publishedby','menutitle','donthit','privateweb','privatemgr','content_dispo','hidemenu','alias_visible');
 $docid = (isset($docid) && (int)$docid>0) ? (int)$docid : $modx->documentIdentifier;
+if(isset($templid) && (int)$templid>0) {
+	$doc['parent'] = $docid;
+	$limit = 20;
+	do {
+		$doc = $modx->getDocument($doc['parent'], 'id, parent, template');
```

Comment:
```$modx->getDocument()``` Can return false. And you work only with arrays...

Reply:
Sorry, but what's the problem with arrays?
I can't use $docid instead $doc['parent'] in $modx->getDocument($doc['parent'],..), because $doc['parent'] will be new with each new iteration.

Reply:
![image](https://user-images.githubusercontent.com/1748872/47366144-7c451080-d6f6-11e8-9df4-17d487673620.png)